sources/specUtils.ts

```diff
@@ -52,6 +53,46 @@ export function parseSpec(raw: unknown, source: string, {enforceExactVersion = t
   };
 }
 
+type CorepackPackageJSON = {
+  packageManager?: string;
+  devEngines?: { packageManager?: DevEngineDependency };
+};
+
+interface DevEngineDependency {
+  name: string;
+  version: string;
+}
+function parsePackageJSON(packageJSONContent: CorepackPackageJSON) {
+  if (packageJSONContent.devEngines?.packageManager) {
+    const {packageManager} = packageJSONContent.devEngines;
+
+    if (Array.isArray(packageManager))
+      throw new UsageError(`Providing several package managers is currently not supported`);
```

Comment:
Perhaps an array of length 1 should be permitted?

Reply:
IMO that can be done as a follow up when adding support for arrays. I’d prefer to keep the implementation simpler for the initial PR

Reply:
Do you mind adding an explanation in the PR description for what you think the eventual complete support looks like? So like that Corepack will eventually support arrays; will it support `onFail` too? Etc.

Reply:
I think the PR description is describing the current PR, I don t think it should describe potential future PRs.

Reply:
Without an explanation, it's unclear how much of the spec Corepack intends to support. And if the support lands piecemeal and is released that way, there might be weird gaps where particular versions of Corepack error on things (like arrays) that other versions don't.

If you flag it or release it all together somehow, I think that's fine; but otherwise maybe just implement all of the spec that you think makes sense for Corepack to implement? The devEngines spec isn't that large.

Reply:
I've changed the throw into a warning